packages/io/svg-serializer/index.js

```diff
@@ -30,40 +27,32 @@ const mimeType = 'image/svg+xml'
  * @param {Object|Array} objects - objects to serialize as SVG
  * @returns {Array} serialized contents, SVG format
  */
-const serialize = (...params) => {
-  let options = {}
-  let objects
-  if (params.length === 0) {
-    throw new Error('no arguments supplied to serialize function !')
-  } else if (params.length === 1) {
-    // assumed to be object(s)
-    objects = Array.isArray(params[0]) ? params[0] : params
-  } else if (params.length > 1) {
-    options = params[0]
-    objects = params[1]
-  }
-  // make sure we always deal with arrays of objects as inputs
-  objects = toArray(objects)
-  console.log('params', params)
-  console.log('options', options)
-  console.log('objects', objects)
+const serialize = (options, ...objects) => {
   const defaults = {
-    statusCallback: null,
     unit: 'mm', // em | ex | px | in | cm | mm | pt | pc
-    decimals: 10000
+    decimals: 10000,
+    statusCallback: null
   }
   options = Object.assign({}, defaults, options)
 
+  objects = utils.flatten(objects)
+
+  // convert only 2D geometries
+  let objects2d = objects.filter((object) => geometry.geom2.isA(object) || geometry.path2.isA(object))
+
+  if (objects2d.length === 0) throw new Error('only 2D geometries can be serialized to SVG')
+  if (objects.length !== objects2d.length) console.warn('some objects could not be serialized to SVG')
+
   options.statusCallback && options.statusCallback({progress: 0})
 
   // get the lower and upper bounds of ALL convertable objects
-  var bounds = getBounds(objects)
+  var bounds = getBounds(objects2d)
```

Comment:
dang, we still have some 'var's perhaps just bulk replace and not waste time for now?

Reply:
NIce catch! Done.